doc/sphinx/docs/python/tutorials/coding_udp_multi_objective.rst

```diff
@@ -0,0 +1,70 @@
+.. _py_tutorial_coding_udp_multi_objective:
+
+Coding a User Defined Problem with constraints multiple objectives
+------------------------------------------------------------------
+
+In this chapter we show how to code an unconstrained user defined problem (UDP) with multiple objectives.
+We assume that the mathematical formulation of problem is the following:
+
+\begin{array}{rl}
+\mbox{minimize:} & f_{1}\left(x\right) = x^{2} \\
+& f_{2}\left(x\right) = \left(x-2\right)^{2} \\
+\mbox{subject to:} & 0 <= x <= 2\\
+\end{array}
+
+which is a test function for multi-objective optimization being introduced in
+*Schaffer, J. David (1984). Some experiments in machine learning using vector
+evaluated genetic algorithms (artificial intelligence, optimization, adaptation,
+pattern recognition) (PhD). Vanderbilt University.* and illustrated
+`here <https://en.wikipedia.org/wiki/Test_functions_for_optimization#Test_functions_for_multi-objective_optimization>`_.
+
+The implementation as UDP can be realized as follows:
+
+.. doctest::
+
+    >>> import pygmo as pg
```

Comment:
no need to import pygmo here. udps can be defined also without pygmo available. You can import pygmo later when using the udp to instantiate a pygmo.problem

Reply:
I have moved the import. See: https://github.com/esa/pagmo2/pull/191/commits/99206785bfb53f8fe36738190f69fcf57f316d57